runtime/src/bank_forks.rs

```diff
@@ -877,16 +877,25 @@ mod tests {
             .collect()
     }
 
+    fn extend_bank_forks(bank_forks: &mut BankForks, parent_child_pairs: &[(Slot, Slot)]) {
+        for (parent, child) in parent_child_pairs.iter() {
+            bank_forks.insert(Bank::new_from_parent(
+                &bank_forks[*parent],
```

Comment:
iirc the index op on bank forks already returns a reference, so I think we may not need the `&` here:

```rust
impl Index<u64> for BankForks {
    type Output = Arc<Bank>;
    fn index(&self, bank_slot: Slot) -> &Self::Output {
        &self.banks[&bank_slot]
    }
}
```

also classic, derefencing the index so we can just reference it

Reply:
Hmm, clippy didn't complain but the about the `&` for `&Arc<Bank>` argument, but that will be going away anyways in the larger PR.

> also classic, derefencing the index so we can just reference it

🎉